### PR TITLE
Update Range.GetOffsetAndLength

### PR DESCRIPTION
**Pull request by tarekgh**
Make it return offset and length tuple instead of the old Type Range.OffsetAndLength.

https://github.com/dotnet/corefx/issues/35508

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @terrajobst @stephentoub @agocke 

This change is independent of the compiler as the compiler is not using it.

File: src/System.Private.CoreLib/shared/System/Range.cs [hunk 1]
```diff
@@ -95,15 +95,15 @@ public override string ToString()
...
+        /// <summary>Calculate the start offset and length of range object using a collection length.</summary>
+        /// <param name="length">The length of the collection that the range will be used with. length has to be a positive value.</param>
         /// <remarks>
         /// For performance reason, we don't validate the input length parameter against negative values.
         /// It is expected Range will be used with collections which always have non negative length/count.
         /// We validate the range is inside the length scope though.
         /// </remarks>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public OffsetAndLength GetOffsetAndLength(int length)
+        public (int offset, int rangeLength) GetOffsetAndLength(int length)
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Range.cs]: The return type in https://github.com/dotnet/corefx/issues/35508 is `(int Offset, int Length)`. Is it intentional that this is different?

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Range.cs]: It should be `int Offset, int Length`

Comment (tarekgh) [reply src/System.Private.CoreLib/shared/System/Range.cs]: I'll fix that.

**Pull request merged**